Bug 1131344 - Fix redundant hover-over-top indicator. r=kgrandon

Cwiiis opened the pull request.

try-server-hook commented:
[<img src="https://avatars.githubusercontent.com/u/668518?v=3" alt="Cwiiis" width="50px" height="50px" /> Chris Lord (Cwiiis)](https://github.com/Cwiiis) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=5a3642bab803)

try-server-hook commented:
[<img src="https://avatars.githubusercontent.com/u/668518?v=3" alt="Cwiiis" width="50px" height="50px" /> Chris Lord (Cwiiis)](https://github.com/Cwiiis) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=ab76536a9446)

KevinGrandon commented on shared/elements/gaia_grid/js/grid_dragdrop.js:
```diff
@@ -489,16 +489,35 @@
           }
+        } else {
+          redundantRearrange = this.gridView.items[foundIndex].element.
+            classList.contains('invalid-drop')
```
;

KevinGrandon commented on shared/elements/gaia_grid/js/grid_dragdrop.js:
```diff
@@ -489,16 +489,35 @@
+        }
+
+        if (redundantRearrange) {
           if (this.hoverItem) {
```
How do you feel about simply removing the ```redundantRearrange``` variable and moving this code into the ```} else {``` statement on line 508? It seems redundant to have an if statement when we could just move the code into the blocks above.

KevinGrandon commented on shared/elements/gaia_grid/js/grid_dragdrop.js:
```diff
@@ -489,16 +489,35 @@
           }
           return;
+        } else if (insertDividerAtTop) {
+          this.container.classList.add('hover-over-top');
```
Same as above, moving this into the if statement on like 507 or so.

Cwiiis replied:
I don't think there's a simplification of the if/else structure that would do away with variables altogether - I'll think about it some, but I'd rather have the consequences grouped together here rather than duplicated and spread over several nested if/elses, if it came down to it.

Cwiiis committed:
Bug 1131344 - Fix redundant hover-over-top indicator. r=kgrandon

Stop showing indications for inserting a group at the top of the container
if the result is redundant.

try-server-hook commented:
[<img src="https://avatars.githubusercontent.com/u/668518?v=3" alt="Cwiiis" width="50px" height="50px" /> Chris Lord (Cwiiis)](https://github.com/Cwiiis) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=b84a1cffd2f1)
